File: csa/lab3/word-counter.js
class WordCounter {
    constructor(src) {
        this.words = src.split(",");
    }

    count() {
        let count = 0;
        let main = this.words[0];

        this.words.forEach(function(word){
            if (word == main){
                count++;
            }
        });

        return count;
    }

    getResult() {
        try {
            return this.count();
        } catch {
            return 0;
        }
    }
}


module.exports = {
    WordCounter,
}

File: csa/lab1/server/src/WordCounter.java
import java.util.ArrayList;
import java.util.Arrays;

public class WordCounter implements Result {

    private ArrayList<String> words = new ArrayList<>();

    public WordCounter(String src) {
        this.words = new ArrayList<String>(Arrays.asList(src.split(",")));
    }

    public WordCounter(ArrayList<String> src) {
        words = src;
    }

    public WordCounter(String ... src) {
        for(int i = 0; i < src.length; i++) {
            this.words.add(src[i]);
        }
    }

    public Integer count(){
        int count = 0;
        String main = this.words.get(0);

        for (String word : this.words){
            if (word.equals(main)){
                count++;
            }
        }

        return count;
    }

    @Override
    public String getResult() {
        return String.valueOf(this.count());
    }
}
